azext_iot/tests/iothub/core/test_iothub_storage_int.py

```diff
@@ -376,3 +402,34 @@ def test_user_identity_storage(self):
             )
 
         self.tearDown()
+
+    def wait_till_job_completion(self, job_id):
+        tries = 0
+        status = self.cmd(
+            f"iot hub job show -n {self.entity_name} -g {self.entity_rg} --job-id {job_id}"
+        ).get_output_in_json()["status"]
+
+        while status not in ["failed", "completed"] and tries < JOB_POLL_MAX_ATTEMPTS:
+            status = self.cmd(
+                f"iot hub job show -n {self.entity_name} -g {self.entity_rg} --job-id {job_id}"
+            ).get_output_in_json()["status"]
+            sleep(SETUP_SLEEP_INTERVAL)
+            tries += 1
+
+        if status == "failed":
+            status = self.cmd(
```

Comment:
We do not need to run the command to check the status again since we already know that the job is failed.

Reply:
I am trying to get the specific error code - will change variable names to reduce confusion.

Reply:
Its better to do something like this:

```
status = "running"
while status not in ["failed", "completed"] and tries < JOB_POLL_MAX_ATTEMPTS:
        status_output = self.cmd(
            f"iot hub job show -n {self.entity_name} -g {self.entity_rg} --job-id {job_id}"
        ).get_output_in_json()
        status = status_output["status"]
        sleep(SETUP_SLEEP_INTERVAL)
        tries += 1

if status == "failed":
    logger.error(status_output)
```